### ISSUE TITLE
Provide a test execution endpoint

### ISSUE THREAD

**Issue by jbonnet** [sonata-nfv/tng-gtk-vnv#23]
1. The Endpoint for the triggering of the tests related to a list of network_service_id's
POST /api/v1/schedulers/services
[{ “network_service_id” : “0011” },
{ “network_service_id” : “0012” },
{ “network_service_id” : “0013” },
…]
1. The Endpoint for the triggering of the services related a list of test_uuid
POST /api/v1/schedulers/tests
[{ “ test_uuid” : “0101”}, { “ test_uuid” : “0102”}, { “ test_uuid” : “0103”}, … ]

**Comment by jbonnet** [comment]
@allemaos @anapolg 
We had this issue covering the execution of a single test (i.e., not triggered by a new package but triggered in the portal)...

Could we have the external API like the following? 
```shell
$ curl -X POST pre-int-vnv-bcn.5gtango.eu:32002/api/v3/tests/descriptors/83020a54-54d1-4931-b6e0-a52918ef0c9f/executions
```

Should the GK use the above mentioned API(`/api/v1/schedulers/services`), or the `curl -v -X POST "http://pre-int-vnv-bcn.5gtango.eu:6100/tng-vnv-lcm/api/v1/packages/on-change" \
-H "Content-Type: application/json" -d '{ "event_name": “ANAUI", "package_id": "'$package_id'"}’ ;`?

**Comment by anapolg** [comment]
@jbonnet  For me the endpoint you are suggesting to be exposed in the GK (http://pre-int-vnv-bcn.5gtango.eu:32002/api/v3/tests/descriptors/83020a54-54d1-4931-b6e0-a52918ef0c9f/executions) is fine.

**Comment by jbonnet** [comment]
An alternative would be , instead of the above proposed API, to have a POST into `…/test/plans`,
the POST’s body would then have `{“service_uuid”:”…”,”test_uuids”:[…]}`. What do you think?
from where is the portal getting those test uuids?

**Comment by jbonnet** [comment]
@anapolg exactely
@allemaos but we need those APIs first (I remember chating with @mrduguo about them)

**Comment by allemaos** [comment]
@jbonnet  cc: @anapolg 

No, GK should not use "/tng-vnv-lcm/api/v1/packages/on-change" for triggering UI's actions.

The endpoints: /api/v1/schedulers/services, /api/v1/schedulers/tests, /api/v1/schedulers was developed for the GK triggering of UI's Execution actions. 

Did I answer your questions?

**Comment by jbonnet** [comment]
Partially, @allemaos.
So is it clear that the V&V's LCM will receive `POST`s like:

* `/api/v1/schedulers/services/<service_uuid>`
* `/api/v1/schedulers/tests/<test_uuid>`

...and proceed accordingly?

**Comment by allemaos** [comment]
@jbonnet internally 

The rule is  
A. The Endpoint for the triggering of the Services( and their related Tests) is
POST /api/v1/schedulers/services
BODY:
[
    { “network_service_id” : “0011” },
    { “network_service_id” : “0012” },
    { “network_service_id” : “0013” },
    …
]
B. The Endpoint for the triggering of the Tests( and their related Services) is
POST /api/v1/schedulers/tests
BODY:
[
    { “ test_uuid” : “0101”}, 
    { “ test_uuid” : “0102”}, 
    { “ test_uuid” : “0103”}, 
    … 
]

C. The Endpoint for the triggering  of the Services( and their related Tests)  and the Tests( and their related Services) is (still in progress) 
POST /api/v1/schedulers
BODY:
[
    [
        { “network_service_id” : “0011” },
        { “network_service_id” : “0012” },
        { “network_service_id” : “0013” },
        ...
     ],
    [
        { “ test_uuid” : “0101”}, 
        { “ test_uuid” : “0102”}, 
        { “ test_uuid” : “0103”}, 
        … 
    ]
]

**Comment by jbonnet** [comment]
Closed via #25, #26 and #27
